core/thegraph/state_integration_test.go

```diff
@@ -123,33 +98,45 @@ func setupTest(t *testing.T) (
 	err = testConfig.DeployExperiment()
 	require.NoError(t, err, "failed to deploy experiment")
 
-	logger.Info("Starting binaries")
-	testConfig.StartBinaries(true)
+	// Create a minimal InfrastructureHarness
+	infraHarness := &inaboxtests.InfrastructureHarness{
+		TestName:   testName,
+		TestConfig: testConfig,
+		Logger:     logger,
+		Ctx:        ctx,
+	}
+
+	// For graph indexer test, we don't need churner functionality
+	churnerURL := ""
+
+	// Start operator nodes as goroutines using the inabox operator setup
+	logger.Info("Starting operator nodes as goroutines")
+	err = inaboxtests.StartOperatorsForInfrastructure(infraHarness, anvilContainerPort, churnerURL)
+	require.NoError(t, err, "failed to start operator nodes")
```

Comment:
are these only spun up for registration? Feel like we should have a separate fct that does that. can you add a todo

Reply:
Added todo 122cd1ee27f09babf5426f5b61d3bccb06675e2c